trino/sqlalchemy/dialect.py

```diff
@@ -0,0 +1,297 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from textwrap import dedent
+from typing import Any, Dict, List, Mapping, Optional, Sequence, Tuple
+
+from sqlalchemy import exc, sql
+from sqlalchemy.engine.base import Connection
+from sqlalchemy.engine.default import DefaultDialect, DefaultExecutionContext
+from sqlalchemy.engine.url import URL
+
+from trino import dbapi as trino_dbapi
+from trino.auth import BasicAuthentication
+from trino.dbapi import Cursor
+from trino.sqlalchemy import compiler, datatype, error
+
+
+class TrinoDialect(DefaultDialect):
+    name = 'trino'
+    driver = 'rest'
+
+    statement_compiler = compiler.TrinoSQLCompiler
+    ddl_compiler = compiler.TrinoDDLCompiler
+    type_compiler = compiler.TrinoTypeCompiler
+    preparer = compiler.TrinoIdentifierPreparer
+
+    # Data Type
+    supports_native_enum = False
+    supports_native_boolean = True
+    supports_native_decimal = True
+
+    # Column options
+    supports_sequences = False
+    supports_comments = True
+    inline_comments = True
+    supports_default_values = False
+
+    # DDL
+    supports_alter = True
+
+    # DML
+    supports_empty_insert = False
+    supports_multivalues_insert = True
+    postfetch_lastrowid = False
+
+    @classmethod
+    def dbapi(cls):
+        """
+        ref: https://www.python.org/dev/peps/pep-0249/#module-interface
+        """
+        return trino_dbapi
+
+    def create_connect_args(self, url: URL) -> Tuple[Sequence[Any], Mapping[str, Any]]:
+        args = list()
```

Comment:
why args is needed here?

Reply:
For return.

Reply:
I see now that it is required return here: https://docs.sqlalchemy.org/en/14/core/internals.html#sqlalchemy.engine.Dialect.create_connect_args